includes/content-distribution/class-incoming-post.php

```diff
@@ -546,19 +560,28 @@ public function insert( $payload = [] ) {
 
 			if ( is_wp_error( $post_id ) ) {
 				self::log( 'Failed to insert post with message: ' . $post_id->get_error_message() );
+
 				return $post_id;
 			}
 
 			// The wp_insert_post() function might return `0` on failure.
 			if ( ! $post_id ) {
 				self::log( 'Failed to insert post.' );
+
 				return new WP_Error( 'insert_error', __( 'Error inserting post.', 'newspack-network' ) );
 			}
 
 			// Update the object.
 			$this->ID   = $post_id;
 			$this->post = get_post( $this->ID );
 
+			Incoming_Author::ingest_author_for_post( $this->ID, $this->get_original_site_url(), $post_data['author'] );
```

Comment:
This method makes a second post update to the db.  Can we dissolve it and move its logic upper in this method's `$postarr` arguments?

Regular WP author is a core aspect of the post and should fit well in the composition of our insert() logic.

Reply:
Makes sense! thanks :)